clippy_lints/src/enum_large_variant.rs

```diff
@@ -0,0 +1,71 @@
+//! lint when there are large variants on an enum
+
+use rustc::lint::*;
+use rustc::hir::*;
+use utils::span_help_and_lint;
+use rustc::ty::layout::TargetDataLayout;
+
+/// **What it does:** Checks for large variants on enums.
+///
+/// **Why is this bad?** Enum size is bounded by the largest variant. Having a large variant
+/// can penalize the memory layout of that enum.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// enum Test {
+///    A(i32),
+///    B([i32; 8000]),
+/// }
+/// ```
+declare_lint! {
+    pub ENUM_LARGE_VARIANT,
```

Comment:
Changed all references to 'enum_large_variant' to 'large_enum_variant' on 088bb8fdb2f4e8fce5a17ea4ae797d3232323b4b